### DISCUSSION TITLE
Apply discount not seen in docs

### DISCUSSION THREAD

**Discussion by SebastianBotez** [lunarphp/lunar#1545] [Q&A] [answered]
What is the method to apply discounts?
I found something digging in tests and got something like this:

```
$discount = Discount::active()->where('handle', $this->couponCode)->first();
if($discount) {
     $cart->coupon_code = $discount->coupon;
     $cart->save();
}
```

The discount column is updating in the database but when I dump the $cart, no discount is applied.
Any help would be highly appreciated as always.

**Comment by glennjacobs** [comment, marked as answer]
The Discount Manager caches the discounts per request, so if you're doing something to the cart that will affect the discounts (like applying a coupon code), you will want to reset the discounts.

E.g.

```php
        $discount = $this->coupon_code;
        if (Discounts::validateCoupon($discount)) {

            $this->cart->coupon_code = $discount;
            $this->cart->save();

            Discounts::resetDiscounts();

        }
```

**Comment by xV4L3x** [comment]
Recalculate the cart

```php
$this->cart->calculate();
```

**Comment by SebastianBotez** [comment]
Thank you guys. Works as a charm! Appreciate your help.
